### PR TITLE
bpf: Send packet drop notify for LB DSR mode failures.

### PR DESCRIPTION
**Pull request by hzhou8**
Packets may be dropped silently for errors such as FIB lookup failure
in DSR mode.  This patch adds drop notify for all errors in DSR functions.

Signed-off-by: Han Zhou <hzhou8@ebay.com>

### CODE REVIEW

**Review by pchaigno — APPROVED** [review]
Thanks!

**Comment by pchaigno** [comment]
test-me-please

**Comment by brb** [comment]
test-runtime

**Review by aditighag — APPROVED** [review]

File: bpf/lib/nodeport.h [hunk 1]
```diff
@@ -1267,24 +1301,30 @@ int tail_nodeport_nat_ipv4(struct __ctx_buff *ctx)
 	if (dir == NAT_DIR_EGRESS) {
 		struct remote_endpoint_info *info;
 
-		if (!revalidate_data(ctx, &data, &data_end, &ip4))
-			return DROP_INVALID;
+		if (!revalidate_data(ctx, &data, &data_end, &ip4)) {
+			ret = DROP_INVALID;
+			goto drop_err;
+		}
```
Comment (borkmann) [inline bpf/lib/nodeport.h:1307]: Hey Han, looks like this one changed the v4 NAT code for the perf event error reporting, but not the corresponding v6 code (`tail_nodeport_nat_ipv6()`). Do you have a chance to follow-up on the v6 as well? Thx

Comment (hzhou8) [reply bpf/lib/nodeport.h:1307]: yes, sorry for the miss. I can send a quick follow up.

**Pull request merged**